### PR TITLE
adding priority-class for esIndexCleaner

### PR DESCRIPTION
**Pull request by swapnilpotnis**
Signed-off-by: Swapnil Potnis <potnis.swapnil@gmail.com>

## Which problem is this PR solving?
- #1620 

## Short description of the changes
- Adding the attribute of "priorityclassName" for "esIndexCleaner"

### CODE REVIEW

**Comment by rubenvp8510** [comment]
The linter is complaining. You need to regenerate the bundle.

You can use the following command.
`VERSION=1.30.0 USER=jaegertracing make bundle`

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : Done.

**Comment by rubenvp8510** [comment]
DCO is failing, You need to sign your commits :)

**Comment by rubenvp8510** [comment]
Hi @swapnilpotnis are you still interesting on this one? If so please rebase and sign your commits I think after that it will be ready to be merged :)

Thanks for your contribution!

**Comment by swapnilpotnis** [comment]
> 

Really sorry for the late response. Will fix it up by today itself and have the PR ready for review.

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : hey...I have added the unit test case as well as updated the PR with the latest branch commit. Can you please take a look?

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : can you please allow the tests to be run over the PR again?

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : the earlier check seemed to have failed with the error "Build failed: the api.md file has been changed but the generated api.md file isn't up to date. Run 'make api-docs' and update your PR." (https://github.com/jaegertracing/jaeger-operator/runs/5548885043?check_suite_focus=true#step:6:171) but running the command states that no rule has been found to the target "api-docs". I am a bit confused as to how to deal with the error and google wasn't helpful as well in that aspect. Can you please provide me some pointers??

**Comment by swapnilpotnis** [comment]
> @rubenvp8510 : the earlier check seemed to have failed with the error "Build failed: the api.md file has been changed but the generated api.md file isn't up to date. Run 'make api-docs' and update your PR." (https://github.com/jaegertracing/jaeger-operator/runs/5548885043?check_suite_focus=true#step:6:171) but running the command states that no rule has been found to the target "api-docs". I am a bit confused as to how to deal with the error and google wasn't helpful as well in that aspect. Can you please provide me some pointers??

@rubenvp8510 : can you please help me with this??

**Comment by rubenvp8510** [comment]
> > @rubenvp8510 : the earlier check seemed to have failed with the error "Build failed: the api.md file has been changed but the generated api.md file isn't up to date. Run 'make api-docs' and update your PR." (https://github.com/jaegertracing/jaeger-operator/runs/5548885043?check_suite_focus=true#step:6:171) but running the command states that no rule has been found to the target "api-docs". I am a bit confused as to how to deal with the error and google wasn't helpful as well in that aspect. Can you please provide me some pointers??
> 
> @rubenvp8510 : can you please help me with this??

Hi did you rebase your PR? try to run it with the PR rebased on top of main. I checkout your branch and I was able to run it and generated the doc file. I can also push a commit with the change if you want.

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : thanks for your help. I have updated the PR with the required changes. Can you please run the checks??

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : Forgive me as I wasn't aware of the naming convention that was suppose to be followed for which the earlier check had failed. Can you please re-run the tests as I have rectified the variable name as part of the latest commit?

**Comment by rubenvp8510** [comment]
> @rubenvp8510 : Forgive me as I wasn't aware of the naming convention that was suppose to be followed for which the earlier check had failed. Can you please re-run the tests as I have rectified the variable name as part of the latest commit?

No problem :)

**Comment by rubenvp8510** [comment]
2022-03-23T14:25:57.3007810Z --- FAIL: TestPriorityClassName (0.00s) :(

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : can you please run the tests for the latest commit ?

**Comment by swapnilpotnis** [comment]
@rubenvp8510 : alas..!!! The pipelines finally Green :)
Can you also please review the PR whenever you have some bandwidth??

**Review by frzifus — COMMENTED** [review]
hi @swapnilpotnis, thanks for your pr. I have only 2 last questions :)

File: pkg/cronjob/es_index_cleaner.go
Comment (rubenvp8510) [reply pkg/cronjob/es_index_cleaner.go:49]: Agree on this comment.

File: apis/v1/jaeger_types.go
Comment (rubenvp8510) [reply apis/v1/jaeger_types.go:686]: This is a k8s concept https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/

File: pkg/cronjob/es_index_cleaner.go
Comment (swapnilpotnis) [reply pkg/cronjob/es_index_cleaner.go:49]: Actually I looked at other parts of the code where priorityClass has been reference and this was the general approach followed elsewhere as well. Having to go with the same flow, I implemented the same set of changes

ref: 
1. https://github.com/jaegertracing/jaeger-operator/blob/main/pkg/deployment/agent.go#L92
2. https://github.com/jaegertracing/jaeger-operator/blob/main/pkg/deployment/query.go#L89

Comment (swapnilpotnis) [reply pkg/cronjob/es_index_cleaner.go:49]: Let me know if you guys wish to have the assignment in the metav1 object itself so that I can make the changes. Also let me know if you guys wish to have the same set of changes made at other places as well. If that be the case, should I do it in the same PR or create a new PR altogether??

Comment (swapnilpotnis) [reply pkg/cronjob/es_index_cleaner.go:49]: @rubenvp8510 / @frzifus : Guys, can you please share your inputs over it?

Comment (frzifus) [reply pkg/cronjob/es_index_cleaner.go:49]: Sorry for the long response. From my point of view it would make sense to change it at least in this pr. But feel free to change it on the other places too in this or a new pr. What do you think?

Comment (swapnilpotnis) [reply pkg/cronjob/es_index_cleaner.go:49]: Sure. Will make the necessary change only for the priorityClass which is part of this PR. 
And would create a new PR for the other 2 PriorityClass.

Comment (rubenvp8510) [reply pkg/cronjob/es_index_cleaner.go:49]: I would say only do the change related to this PR, we can do other changes in a separate PR

**Comment by swapnilpotnis** [comment]
@rubenvp8510 / @frzifus : Can you guys please review my PR?

File: pkg/cronjob/es_index_cleaner.go
Comment (frzifus) [reply pkg/cronjob/es_index_cleaner.go:49]: still a bit confused, isnt it the same like?
```suggestion
	priorityClassName := "jaeger.Spec.Storage.EsIndexCleaner.PriorityClassName"
```

up to you @rubenvp8510

Comment (rubenvp8510) [reply pkg/cronjob/es_index_cleaner.go:49]: I agree with the change, but for not blocking this PR we can do the changes in a follow up PR.

**Review by rubenvp8510 — APPROVED** [review]

**Comment by swapnilpotnis** [comment]
@rubenvp8510: thanks alot :)
will soon create a PR for the priorityClass change at other 2 places

**Pull request merged**
